packages/example-codehub/src/codehub-application.ts

```diff
@@ -3,16 +3,19 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {Application, CoreBindings} from '@loopback/core';
+import {Application} from '@loopback/core';
+import {RestServer, RestBindings} from '@loopback/rest';
 import {UserController, HealthController} from './controllers';
 
 export class CodeHubApplication extends Application {
   constructor() {
     super();
 
     const app = this;
+    const server = new RestServer(this, {});
+    app.server(server);
 
-    app.api({
+    server.api({
```

Comment:
This is interesting. IMO, it makes more sense to define the REST API at application level, because that's the place where the controllers are defined too. Then we can have multiple servers (e.g. http/https servers listening on different ports) handling incoming request for the same app (same REST API).

Reply:
The `api` function as it's currently written expects an `OpenApiSpec` object; if we move this to the app, then we have to alter the function to require something more permissive.

Reply:
This is the api function right now: 
```ts
  api(spec: OpenApiSpec): Binding {
    return this.bind(RestBindings.API_SPEC).to(spec);
  }
```
I don't see how we can decouple it without unravelling the value of the sugar it provides.

Reply:
I see. I agree it makes sense to keep the implementation of this in the rest component.

Can we add `app.api` via a mixin, similarly to how we are adding `app.repository`?